src/App.tsx

```diff
@@ -1,14 +1,35 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { actions as todosActions } from './features/todos';
+import { useAppDispatch, useAppSelector } from './app/hooks';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    setLoading(true);
+
+    setTimeout(() => {
+      getTodos()
+        .then(data => dispatch(todosActions.setTodos(data)))
+        .catch(() => setErrorMessage('There are no todos'))
+        .finally(() => setLoading(false));
+    }, 1000);
+  }, []);
```

Comment:
```suggestion
  }, [dispatch]);
```
it's better to just set it - FYI
![image](https://github.com/mate-academy/react_redux-list-of-todos/assets/86923089/dcac3b4b-c8ca-41fc-bbe7-4048f3a73055)

dispatch is also memorized under the hood (same for setState functions, etc.) and never change, so the useEffect will also work only once.